test/ability_leaf_guard.c

```diff
@@ -0,0 +1,48 @@
+#include "global.h"
+#include "test_battle.h"
+
+SINGLE_BATTLE_TEST("Leaf Guard prevents non-volatile status conditions in sun")
```

Comment:
```suggestion
SINGLE_BATTLE_TEST("Leaf Guard prevents burns from moves in sun")
```

Reply:
It's written like this in other tests I looked at, but it doesn't really matter imo.

`SINGLE_BATTLE_TEST("Misty Terrain protects grounded battlers from non-volatile status conditions")`

Reply:
As of the latest commit, this test now checks for every status using will-o-wisp, hypnosis, thunder wave, toxic, and powder snow.

Let me know if you still want the label changed.

Reply:
Ah, if that's the case then the label you had before now fits :)